### ISSUE TITLE
Inconsistent behavior of cppcoreguidelines-noexcept-move-operations 

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#76159]
Hi,

Consider this example:
https://godbolt.org/z/5zK9abTjz

```cpp
#include <fstream>

struct Data
{
    Data() = default;

    Data(Data&&)            = default;
    Data& operator=(Data&&) = default;

    Data(Data const&)            = delete;
    Data& operator=(Data const&) = delete;

    ~Data() { }

 private:
    std::ofstream file_{};
};
```
Here, clang-tidy warnings about the operations. However, if I remove the entire private section, then clang-tidy does not warn.

Honestly I don't know what the correct behavior should be. Personally I think all `= default` operations should probably not be warned about. 

Anyway, this warning should have a consistent behavior regardless of whether the struct has a private section or not.

**Comment by PiotrZSL** [comment]
@carlosgalvezp 
Check is correct here. This is because std::ostream has non noexcept move constructor/assign operator.
So check is detecting that this move constructor will be actually "noexcept-unevaluated" and ask for an explicit noexcept.
